src/test/java/games/strategy/engine/framework/startup/login/ClientLoginValidatorTests.java

```diff
@@ -92,9 +115,6 @@ public void shouldReturnChallengeIgnoredByHmacSha512Authenticator() {
     @InjectMocks
     private ClientLoginValidator clientLoginValidator;
 
-    @Mock
-    private IServerMessenger serverMessenger;
```

Comment:
Good catch on these.  They must have been leftovers from when we were testing via the public `verifyConnection()` method (which uses `serverMessenger`) before we extracted the package-private `authenticate()` method (which doesn't use `serverMessenger`).  However, by removing this mock, we are implicitly injecting a `null` value for `serverMessenger` via the `ClientLoginValidator` constructor.

IMO, it would be best to make this explicit.  Therefore, can you replace:

```java
@InjectMocks
private ClientLoginValidator clientLoginValidator;
```

with

```java
private final ClientLoginValidator clientLoginValidator = new ClientLoginValidator(null);
```

And a similar change in `GetChallengePropertiesTests` above?  You can then remove the unused `InjectMocks` import.  I ran the tests with those changes and they continue to pass.

Reply:
Good suggestions, though I think at the moment I would prefer to leave this out of scope. I was merely trying to add to this test case and there were a few opportunistic cleanups, the goal of this PR is *not* to fix up test code. Any objection if we tackle these on a new branch?

Reply:
Not at all.  I'll do that after this PR is merged.